dev/managedservicesplatform/internal/resource/monitoringalertpolicy/monitoringalertpolicy.go

```diff
@@ -0,0 +1,299 @@
+		}
+		return thresholdAggregation(scope, id, config)
+	}
+	return responseCodeMetric(scope, id, config)
+}
+
+// threshholdAggregation defines a monitoring alert policy based on a single metric threshold
+func thresholdAggregation(scope constructs.Construct, id resourceid.ID, config *Config) (*Output, error) {
+	// Set some defaults
+	if config.ServiceKind == spec.ServiceKindService {
+		// For Cloud Run Services we need to group by the revision_name
+		config.ThresholdAggregation.GroupByField = "resource.label.revision_name"
+	} else if config.ServiceKind == spec.ServiceKindJob {
+		// No default for this
+	} else {
+		return nil, errors.Newf("invalid service kind %q", config.ServiceKind)
+	}
+
+	if config.ThresholdAggregation.Comparison == "" {
+		config.ThresholdAggregation.Comparison = ComparisonGT
+	}
+
+	if config.ThresholdAggregation.Duration == "" {
+		config.ThresholdAggregation.Duration = "0s"
+	}
+
+	// Optional on Jobs
+	var group_by *[]*string
+	if config.ThresholdAggregation.GroupByField != "" {
+		group_by = pointers.Ptr([]*string{pointers.Ptr(config.ThresholdAggregation.GroupByField)})
+	}
+
+	_ = monitoringalertpolicy.NewMonitoringAlertPolicy(scope,
+		id.TerraformID(config.ID), &monitoringalertpolicy.MonitoringAlertPolicyConfig{
+			Project:     pointers.Ptr(config.ProjectID),
+			DisplayName: pointers.Ptr(config.Name),
+			Documentation: &monitoringalertpolicy.MonitoringAlertPolicyDocumentation{
+				Content:  config.Description,
+				MimeType: pointers.Ptr("text/markdown"),
+			},
+			Combiner: pointers.Ptr("OR"),
+			Conditions: []monitoringalertpolicy.MonitoringAlertPolicyConditions{
+				{
+					DisplayName: pointers.Ptr(config.Name),
+					ConditionThreshold: &monitoringalertpolicy.MonitoringAlertPolicyConditionsConditionThreshold{
+						Aggregations: []monitoringalertpolicy.MonitoringAlertPolicyConditionsConditionThresholdAggregations{
+							{
+								AlignmentPeriod:    pointers.Ptr(config.ThresholdAggregation.Period),
+								PerSeriesAligner:   pointers.Ptr(string(config.ThresholdAggregation.Aligner)),
+								CrossSeriesReducer: pointers.Ptr(string(config.ThresholdAggregation.Reducer)),
+								GroupByFields:      group_by,
+							},
+						},
+						Comparison:     pointers.Ptr(string(config.ThresholdAggregation.Comparison)),
+						Duration:       pointers.Ptr(config.ThresholdAggregation.Duration),
+						Filter:         pointers.Ptr(filter(config)),
+						ThresholdValue: pointers.Float64(config.ThresholdAggregation.Threshold),
+						Trigger: &monitoringalertpolicy.MonitoringAlertPolicyConditionsConditionThresholdTrigger{
+							Count: pointers.Float64(1),
+						},
+					},
+				},
+			},
+			AlertStrategy: &monitoringalertpolicy.MonitoringAlertPolicyAlertStrategy{
+				AutoClose: pointers.Ptr("604800s"),
+			},
+		})
+	return &Output{}, nil
+}
+
+// filter creates the Filter string for a ThresholdAggregation monitoring alert policy
+func filter(config *Config) string {
```

Comment:
nit - we use `filters` as a variable name, let's make this a little more explicit:

```suggestion
// buildFilter creates the Filter string for a ThresholdAggregation monitoring alert policy
func buildFilter(config *Config) string {
```